packages/pluggableWidgets/badge-web/src/components/__tests__/Badge.spec.tsx

```diff
@@ -0,0 +1,72 @@
+import { shallow, ShallowWrapper } from "enzyme";
+import { createElement } from "react";
+
+import { Badge, BadgeProps } from "../Badge";
+
+describe("Badge", () => {
+    const createBadge = (props: BadgeProps): ShallowWrapper<any, any> => shallow(<Badge {...props} />);
+
+    let defaultBadgeProps: BadgeProps;
+
+    beforeEach(() => {
+        defaultBadgeProps = {
+            type: "badge",
+            value: "text"
+        };
+    });
+
+    it("renders as a badge", () => {
+        const badge = createBadge(defaultBadgeProps);
+
+        expect(badge).toMatchSnapshot();
+    });
+
+    it("renders as a label", () => {
+        defaultBadgeProps.type = "label";
+        const badge = createBadge(defaultBadgeProps);
+
+        expect(badge).toMatchSnapshot();
+    });
+
+    it("renders when an empty string is passed as value", () => {
+        defaultBadgeProps.value = "";
+        const badge = createBadge(defaultBadgeProps);
+
+        expect(badge).toMatchSnapshot();
+    });
+
+    it("renders as a button like element when onClick function is passed", () => {
```

Comment:
can you add a test case simulating triggering the event and checking if the function was called? Same applies for onKeyDown

Reply:
Done :)